core/configuration-service/src/main/java/org/hawkular/metrics/sysconfig/ConfigurationService.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.sysconfig;
+
+import static com.datastax.driver.core.BatchStatement.Type.UNLOGGED;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hawkular.rx.cassandra.driver.RxSession;
+
+import com.datastax.driver.core.BatchStatement;
+import com.datastax.driver.core.PreparedStatement;
+
+import rx.Observable;
+
+/**
+ * @author jsanda
+ */
+public class ConfigurationService {
+
+    private RxSession session;
+
+    private PreparedStatement findConfiguration;
+
+    private PreparedStatement updateConfiguration;
+
+    // TODO make async
+    // I could have just as easily passed the session as a constructor arg. I am doing it in the init method because
+    // eventually I would like service initialization async.
+    public void init(RxSession session) {
+        this.session = session;
+        findConfiguration = session.getSession().prepare("SELECT name, value FROM sys_config WHERE config_id = ?");
+        updateConfiguration = session.getSession().prepare(
+                "INSERT INTO sys_config (config_id, name, value) VALUES (?, ?, ?)");
+    }
+
+    public Observable<Configuration> load(String id) {
```

Comment:
The Single class is marked experimental which presumably means it could go away in future releases. There are other places where this would be good to use like on inserts where we return Observable<Void>, but do we want to depend on experimental features that may go away?